### PR TITLE
EN-2980, EN-3261,  support sensitive routes and pass reason

### PR DESCRIPTION
**Pull request by nitzanpx**

### CODE REVIEW

File: PerimeterXModule/Internals/Validators/PXS2SValidator.cs [hunk 1]
```diff
@@ -20,6 +21,7 @@ public PXS2SValidator(PxModuleConfigurationSection PxConfig, HttpClient HttpClie
 
 		public bool VerifyS2S(PxContext PxContext)
 		{
+			var riskRttStart = Stopwatch.StartNew();
```
Comment (yaronpx) [inline PerimeterXModule/Internals/Validators/PXS2SValidator.cs]: Make sure you stop the the Stopwatch before exiting this method

File: PerimeterXModule/DataContracts/Activities/ActivityDetails.cs [hunk 2]
```diff
@@ -16,6 +16,11 @@ public class ActivityDetails
 
 		[DataMember(Name = "block_score")]
 		public double RiskScore;
-	}
 
+		[DataMember(Name = "pass_reason", EmitDefaultValue = true)]
+		public PassReasonEnum? PassReason;
+
+		[DataMember(Name = "risk_rtt")]
+		public double RiskRoundtripTime;
```
Comment (yaronpx) [inline PerimeterXModule/DataContracts/Activities/ActivityDetails.cs]: You eventually assign a `long` value to this property (`Stopwatch.ElapsedMilliseconds`), so save the implicit casting and declare it as `long` in the first place.

While I'm nitpicking your type decisions, I see that `RiskScore` is also `double`, when  it should actually be `int` (even `short` or `byte` will do for a number between 0-100).

File: PerimeterXModule/Internals/Validators/PXS2SValidator.cs (hunk 1 above)
Comment (nitzanpx) [inline PerimeterXModule/Internals/Validators/PXS2SValidator.cs]: fixed function flow, stopping watch at the end of the funciton

**Comment by nitzanpx** [comment]
@yaronpx  please re-review this pr

**Comment by nitzanpx** [comment]
@yaronpx  if everything looks ok please merge

**Pull request merged**